test/odbc/psqlodbc/time.cpp

```diff
@@ -0,0 +1,575 @@
+  dropObject(ServerType::MSSQL, "TABLE", BBF_TABLE_NAME);
+}
+
+// BBF 
+//    Supports 7th digit fractional second
+//    pads fractional seconds with 0s
+//    Length is always 16 (due to padding)
+// PG 
+//    Rounds 7th digit fractional second
+//    does NOT pad
+//    length is dynamic due to not having automatic padding
+TEST_F(PSQL_DataTypes_Time, Insertion_Success) {
+  vector<string> inserted_values = {
+    "NULL",
+    "00:00:00",         // Min
+    "12:34:56",
+    // "23:59:59.1",
+    // "23:59:59.22",
+    // "23:59:59.333",
+    // "23:59:59.4444",
+    // "23:59:59.55555",
+    // "23:59:59.666666",
+    // BBF Supports fractional seconds to 7 places
+    // PG Rounds up the 7th digit
+    // "23:59:59.7777777",  // Max
+  };
+  const int NUM_OF_DATA = inserted_values.size();
+
+  vector<string> expected_values = getExpectedResults(inserted_values);
+
+  createTable(ServerType::MSSQL, BBF_TABLE_NAME, TABLE_COLUMNS);
+
+  testInsertionSuccess(ServerType::MSSQL, BBF_TABLE_NAME, COL1_NAME, inserted_values, expected_values);
+  insertValuesInTable(ServerType::PSQL, PG_TABLE_NAME, inserted_values, false, NUM_OF_DATA);
+
+  inserted_values = duplicateElements(inserted_values);
+  expected_values = duplicateElements(expected_values);
+
+  verifyValuesInObject(ServerType::PSQL, PG_TABLE_NAME, COL1_NAME, inserted_values, inserted_values);
+  verifyValuesInObject(ServerType::MSSQL, BBF_TABLE_NAME, COL1_NAME, inserted_values, expected_values);
+
+  dropObject(ServerType::PSQL, "TABLE", PG_TABLE_NAME);
+  dropObject(ServerType::MSSQL, "TABLE", BBF_TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Time, Insertion_Fail) {
+  RETCODE rcode;
+  OdbcHandler BBF_odbcHandler(Drivers::GetDriver(ServerType::MSSQL));
+  OdbcHandler PG_odbcHandler(Drivers::GetDriver(ServerType::PSQL));
+
+  const vector<string> INVALID_INSERTED_VALUES = {
+    // "24:00:0",    // Hour over range
+    "00:60:0",    // Minute over range
+    // "00:00:60",   // Second over range
+    "00-00-00",   // Format hh-mm-ss
+
+    // ODBC API format for Time is hh:mm:ss
+    // Below are valid in SQL Server
+    "00:00:00.00 AM"
+    "00AM"
+    "00 AM"
+  };
+
+  createTable(ServerType::MSSQL, BBF_TABLE_NAME, TABLE_COLUMNS);
+
+  testInsertionFailure(ServerType::MSSQL, BBF_TABLE_NAME, COL1_NAME, INVALID_INSERTED_VALUES, false);
+  // testInsertionFailure(ServerType::PSQL, PG_TABLE_NAME, COL1_NAME, INVALID_INSERTED_VALUES, false);
+
+  dropObject(ServerType::MSSQL, "TABLE", BBF_TABLE_NAME);
+  dropObject(ServerType::PSQL, "TABLE", PG_TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Time, Update_Success) {
+  const vector<string> INSERTED_VALUES = {
+    "00:00:00"
+  };
+  const vector<string> EXPECTED_VALUES = getExpectedResults(INSERTED_VALUES);
+
+  const vector<string> UPDATED_VALUES = {
+    "12:34:56",
+    // "23:59:59.1",
+    // "23:59:59.22",
+    // "23:59:59.333",
+    // "23:59:59.4444",
+    // "23:59:59.55555",
+    // "23:59:59.666666",
+    "00:00:00"
```

Comment:
Could we test updating the value to NULL?

Reply:
Will have to wait until #718 gets merged in as changes to support `NULL` for updates are in there.

Reply:
Done.